go/array_string/min_window/solution_test.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import "testing"
+
+func TestMinWindow(t *testing.T) {
+	cases := []struct {
+		name, mainString, substring, want string
+	}{
+		{"minimum substring with repeated character", "ADOBECODEBANC", "OBOC", "OBECO"},
+		{"minimum substring with unique character", "ADOBECODEBANC", "ABC", "BANC"},
+		{"use string input without match", "a", "b", ""},
+		{"use empty string inputs", "", "", ""},
+		{"give length of substring less than main string", "ABC", "ADOBECODEBANC", ""},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := minWindow(tc.mainString, tc.substring)
+
+			if got != tc.want {
+				t.Errorf("minWindow(%s, %s) got %v; want %s.", tc.mainString, tc.substring, got, tc.want)
```

Comment:
If you change all of the formatting verbs here to `%q` then it will quote the string and make it easier for you to debug if you have an empty string. In practice you may not be the person debugging your own code, in which case it's even more important that the error message properly communicates your intent to the other engineer (who may actually be you so far in the future that you don't remember that it's an empty string)

Reply:
Awesome, thank you!